Code/src/main/java/nl/utwente/group10/ui/components/blocks/output/ValueBlock.java

```diff
@@ -73,24 +69,20 @@ public final String getValue() {
     public final StringProperty valueProperty() {
         return value;
     }
-
+    
     @Override
-    public Expr asExpr() {
-        // TODO: support more types than floats
-        return new Value(new ConstT("Float"), getValue());
-    }
-
-    @Override
-    public Type getOutputSignature(Env env) {
-        try {
-            return asExpr().analyze(env);
-        } catch (HaskellException e) {
-            throw new TypeUnavailableException();
-        }
+    public void updateExpr() {
+        this.expr = new Value(new ConstT("Float"), getValue()); 
```

Comment:
Might be a GitHub diff view quirk, but this seems to reverse the changes in `valdefblock` somehow.